client/my-sites/themes/themes-site-selector-modal.jsx

```diff
@@ -13,61 +11,73 @@ import partial from 'lodash/function/partial';
 import Theme from 'components/theme';
 import SiteSelectorModal from 'components/site-selector-modal';
 import Helper from 'lib/themes/helpers';
+import i18n from 'lib/mixins/i18n';
 
 const ThemesSiteSelectorModal = React.createClass( {
 	propTypes: {
 		selectedAction: PropTypes.string.isRequired,
 		selectedTheme: PropTypes.object.isRequired,
 		onHide: PropTypes.func,
-		actions: PropTypes.object,
-		getOptions: PropTypes.func
+		actions: PropTypes.object
 	},
 
-	setSiteAndAction( action, theme, site ) {
+	redirectAndCallAction( site ) {
 		/**
 		 * Since this implies a route change, defer it in case other state
 		 * changes are enqueued, e.g. setSelectedTheme.
 		 */
 		defer( () => {
+			const {
+				selectedAction: action,
+				selectedTheme: theme
+			} = this.props;
 			Helper.trackClick( 'site selector', action );
 			page( '/design/' + site.slug );
 			this.props.actions[ action ]( theme, site );
 		} );
 	},
 
-	getUrl( action, theme, site, getOptions ) {
-		var option = find( getOptions( site, theme ), { name: action } );
-
-		if ( option ) {
-			return option.url;
-		}
+	getActionText( name ) {
```

Comment:
To de-dupe this for now we could move it back to the `theme-options` module as a separate function (maybe adding the rest of the strings) and have `getButtonOptions()` call it as well.